akka-http-tests/src/test/scala/akka/http/scaladsl/server/directives/BasicDirectivesSpec.scala

```diff
@@ -107,4 +108,35 @@ class BasicDirectivesSpec extends RoutingSpec {
       } ~> check { responseAs[String] shouldEqual "" }
     }
   }
-}
\ No newline at end of file
+
+  "The `extractStrictEntity` directive" should {
+    "change request to contain strict entity for inner routes" in {
+      val chunks = () ⇒ List("Akka", "HTTP").map(HttpEntity.Chunk(_)).iterator
+      val entity = HttpEntity.Chunked(ContentTypes.`text/plain(UTF-8)`, Source.fromIterator(chunks))
+
+      Post("/abc", entity) ~> {
+        extractRequestEntity { before ⇒
+          extractStrictEntity(200.millis) { _ ⇒
+            extractRequestEntity { after ⇒
+              complete(Seq(before, after).map(_.isInstanceOf[HttpEntity.Strict]).mkString(" => "))
+            }
+          }
+        }
+      } ~> check { responseAs[String] shouldEqual "false => true" }
+    }
+
+    "only consume data once when nested" in {
+      val random = java.util.concurrent.ThreadLocalRandom.current
+      val chunks = Source.fromIterator(() ⇒ Iterator.continually(random.nextInt(0, 2).toString).take(100).map(HttpEntity.Chunk(_)))
```

Comment:
That's unsafe as the block using `random` is not guaranteed to run in the same thread as the surrounding scope. Maybe make it a `def random`.